src/components/Button/Button.tsx

```diff
@@ -70,11 +70,11 @@ export const Button = forwardRef(
       <Element
         autoFocus={autoFocus}
         className={classNames}
-        disabled={disabled}
+        disabled={isDisabled}
         to={href}
         id={id}
         onClick={(e: ReactMouseEvent) => {
-          if (onClick && !disabled) {
+          if (onClick && !isDisabled) {
```

Comment:
I know this isn't part of your PR, but as you've already changed this line could you check the `typeof onClick === 'function'`, rather than checking that it is truthy please? 

That will guard better against (e.g.) someone providing the `onClick` prop as a string.